### PR TITLE
fix(grpc): compute service name dynamically

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The service name can be changed post-patch-time. This patch adds the
functionality to look-up the service name dynamically to achieve the correct
service name behaviour.

### CODE REVIEW

File: ddtrace/utils/__init__.py [hunk 1]
```diff
@@ -28,10 +28,17 @@ def __get__(self, cls, owner):
...
+    """Compute the service name that should be used for an integration
+    based off the given config and pin instances.
+    """
     if pin.service:
         return pin.service
-    elif config.service:
+
+    # Integrations unfortuantely use both service and service_name in their
+    # configs :/
+    elif "service" in config:
```
Comment (brettlangdon) [inline ddtrace/utils/__init__.py:39]: hey, great catch here

File: ddtrace/utils/__init__.py [hunk 2]
```diff
@@ -28,10 +28,17 @@ def __get__(self, cls, owner):
...
 
 def integration_service(config, pin):
+    """Compute the service name that should be used for an integration
+    based off the given config and pin instances.
+    """
     if pin.service:
         return pin.service
-    elif config.service:
+
+    # Integrations unfortuantely use both service and service_name in their
```
Comment (jd) [inline ddtrace/utils/__init__.py:37]:
```suggestion
    # Integrations unfortunately use both service and service_name in their
```

File: ddtrace/contrib/grpc/patch.py [hunk 3]
```diff
@@ -15,12 +17,23 @@
...
+if "DD_GRPC_SERVICE" in os.environ:
+    service = os.getenv("DD_GRPC_SERVICE")
+elif config._get_service():
+    service = "{}-{}".format(config._get_service(), constants.GRPC_SERVICE_CLIENT)
+else:
+    service = constants.GRPC_SERVICE_CLIENT
+
+
 # TODO[tbutt]: keeping name for client config unchanged to maintain backwards
 # compatibility but should change in future
```
Comment (majorgreys) [inline ddtrace/contrib/grpc/patch.py:34]: Can we safely remove this now?

Comment (Kyle-Verhoog) [reply ddtrace/contrib/grpc/patch.py:34]: Hmm. I think you were maybe meaning that the config name itself should be "grpc-client" (as the server is "grpc-server")?

Comment (Kyle-Verhoog) [reply ddtrace/contrib/grpc/patch.py:34]: (In which case the TODO still applies.)

Comment (majorgreys) [reply ddtrace/contrib/grpc/patch.py:34]: Good thing I made that super clear in my todo 🤦

**Pull request merged**